src/server/game/Entities/Creature/SummonInfo/SummonInfo.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "SummonInfo.h"
+#include "Creature.h"
+#include "DB2Stores.h"
+#include "ObjectAccessor.h"
+#include "SummonInfoArgs.h"
+
+SummonInfo::SummonInfo(Creature* summonedCreature, SummonInfoArgs const& args) : _summonedCreature(summonedCreature)
```

Comment:
I would split the constructor into correct initialisation of members and initialisation of despawn flags:

```suggestion
SummonInfo::SummonInfo(Creature* summonedCreature, SummonInfoArgs const& args)
    : _summonedCreature(summonedCreature),
      _remainingDuration(args.Duration),
      _summonerGUID(args.SummonerGUID)
{
    InitializeDespawnFlags(args.SummonPropertiesID);
}

void SummonInfo::InitializeDespawnFlags(Optional<uint32> summonPropertiesID)
{
    uint32 propertiesID = summonPropertiesID.value_or(0);
    if (SummonPropertiesEntry const* summonProperties = sSummonPropertiesStore.LookupEntry(propertiesID))
    {
    
    auto flags = summonProperties->GetFlags();
    _shouldDespawnOnSummonerLogout = flags.HasFlag(SummonPropertiesFlags::DespawnOnSummonerLogout);
    _shouldDespawnOnSummonerDeath = flags.HasFlag(SummonPropertiesFlags::DespawnOnSummonerDeath);
    _shouldDespawnWhenExpired = flags.HasFlag(SummonPropertiesFlags::DespawnWhenExpired);
    }
}
```

Reply:
then auto might get Shauren-banned, but huh

Reply:
The initialization of these settings will get expanded once I get down to implement them. There are a couple more checks that must be performed to correctly assign these settings. For example totem slot summons should always despawn when the summoner dies. But this has to wait until I get to summon slots